src/Microsoft.Crank.AzureDevOpsWorker/Program.cs

```diff
@@ -107,10 +107,12 @@ private static async Task MessageHandler(ProcessMessageEventArgs args)
 
                 if (records == null)
                 {
-                    Console.ForegroundColor = ConsoleColor.DarkRed;
-                    Console.WriteLine($"{LogNow} Could not retrieve records...");
-                    Console.ResetColor();
+                    await devopsMessage?.SendTaskCompletedEventAsync(DevopsMessage.ResultTypes.Skipped);
 
+                    Console.WriteLine($"{LogNow} Could not retrieve Records, skipping...");
+
+                    // Release the message for further processing
+                    await args.AbandonMessageAsync(message);
```

Comment:
REST API failure, abandon the message such that the next loop can try it again

Reply:
We're abandoning so that service bus will retry it, but telling AzDo that the job is skipped?

Reply:
Yes, because we told AzDo to "start" it so the blue icon is spinning (it's fake it's a server-less job) so to stop it we make it skipped, then the next time it's taken it will be marked running again (spinning blue) and eventually fail (red) or (success with issues, orange)

Reply:
I guess I don't understand the flow here. Why does AzDo run it again?

Reply:
AzDo runs a PublishServiceBus task, which only generates a message in the queue and doesn't update the task itself (white circle). Then as we read them the message contains a callback url to update the task in azdo with state (completed, status, log, ...). As we run the benchmark we mark it "running", then "completed" with a status (failed, success, warning).

Reply:
Ok, so here we update it with skipped. But then we reprocess the message because we abandoned it and service bus resends it? Can you then update a previously completed AzDo task?

Reply:
I think so, the REST endpoint is the same for start/complete(+status): a POST to `{PlanUrl}/{ProjectId}/_apis/distributedtask/hubs/{HubName}/plans/{PlanId}/events`. There is nothing that tells you can't change from one state to another (and there are other ones than these two)

If only I could find some doc :/